templates/rbac.yaml

```diff
@@ -55,6 +55,10 @@ roleRef:
   name: {{ .Release.Namespace }}
 subjects:
 - kind: ServiceAccount
+  {{- if .Values.serviceAccount.name}}
+  name: {{ .Values.serviceAccount.name }}
+  {{- else }}
   name: {{ .Release.Namespace }}
+  {{- end }}
```

Comment:
Thank you for the fix. 
If I understand correctly, the issue is that this name must be identical to the service name. Would it make sense to extract this as a shared helper then, such as

```
{{- define "helpers.serviceName" -}}
{{- if .Values.serviceAccount.name}}
{{ .Values.serviceAccount.name }}
{{- else }}
{{ .Release.Namespace }}
{{- end }}
{{- end }}
```

and then use it in both places, here and the service definition?